dashboard/django/stats/views.py

```diff
@@ -282,3 +305,55 @@ def clusters(request):
     return render(request, 'stats/clusters.html', {
         'clusters' : clusters,
     })
+
+
+def clearSubscription(request, topic_name, subscription_name):
+    url = settings.SERVICE_URL + '/admin/v2/' + topic_name + '/subscription/' + subscription_name + '/skip_all'
+    requests.post(url)
+    return redirect('topic', topic_name=topic_name)
+
+def deleteSubscription(request, topic_name, subscription_name):
+    url = settings.SERVICE_URL + '/admin/v2/' + topic_name + '/subscription/' + subscription_name
+    response = requests.delete(url)
+    status = response.status_code
+    if status == 204:
+        ts = get_timestamp()
+        topic_db_name = 'persistent://' + topic_name.split('persistent/', 1)[1]
+        topic = Topic.objects.filter(name=topic_db_name, timestamp=ts)[0]
+        Subscription.objects.filter(name=subscription_name, topic=topic, timestamp=ts).update(deleted=True)
+        subscriptions = Subscription.objects.filter(topic=topic, deleted=False, timestamp=ts)
+        if not subscriptions:
+            topic.deleted=True
+            topic.save(update_fields=['deleted'])
+            m = re.search(r"persistent/(?P<namespace>.*)/.*", topic_name)
+            namespace_name = m.group("namespace")
+            return redirect('namespace', namespace_name=namespace_name)
+    return redirect('topic', topic_name=topic_name)
+
+def messages(request, topic_name, subscription_name):
+    topic_name = 'persistent://' + topic_name.split('persistent/', 1)[1]
+    timestamp = get_timestamp()
+    cluster_name = request.GET.get('cluster')
+
+    if cluster_name:
+        topic = get_object_or_404(Topic, name=topic_name, cluster__name=cluster_name, timestamp=timestamp)
+    else:
+        topic = get_object_or_404(Topic, name=topic_name, timestamp=timestamp)
+    subscription = get_object_or_404(Subscription, topic=topic, name=subscription_name)
+
+    return render(request, 'stats/messages.html', {
+        'topic' : topic,
+        'subscription' : subscription,
+        'title' : topic.name,
+        'subtitle' : subscription_name,
+    })
+
+def peek(request, topic_name, subscription_name, message_number):
+    url = settings.SERVICE_URL + '/admin/v2/' + topic_name + '/subscription/' + subscription_name + '/position/' + message_number
```

Comment:
This option settings.SERVICE_URL needs to be included in the configuration file settings.

Reply:
Done